raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -2025,3 +2025,218 @@ def test_mediator_lock_expired_with_receive_lock_expired():
+    )
+
+    lock_expired_state_change = ReceiveLockExpired(
+        HOP1,
+        balance_proof,
+        transfer.lock.secrethash,
+        1,
+    )
+
+    iteration = mediator.state_transition(
+        iteration.new_state,
+        lock_expired_state_change,
+        channel_map,
+        pseudo_random_generator,
+        block_lock_expired + 1,
+    )
+
+    # LockExpired should remove the lock from both lockedlocks and unlockedlocks
+    assert secrethash not in payer_channel.partner_state.secrethashes_to_lockedlocks
+    assert secrethash not in payer_channel.partner_state.secrethashes_to_unlockedlocks
+
+
+def test_mediator_lock_expired_after_receive_secret_reveal():
+    """
+    Test the following scenario:
+    I -> M -> T
+
+    - Mediator forwards a transfer from I to T.
+    - T receives secret from I and never registers into on-chain.
+    - T sends SecretReveal to M, lockedlock becomes an unlockedlock
+    - M expires the lock.
+    - M should remove the lock from both secrethashes_to_lockedlocks
+      and secrethashes_to_unlocklocks.
+    """
+    amount = 10
+    block_number = 5
+    target = HOP2
+    expiration = 30
+    pseudo_random_generator = random.Random()
+
+    payer_channel = factories.make_channel(
+        partner_balance=amount,
+        partner_address=UNIT_TRANSFER_SENDER,
+        token_address=UNIT_TOKEN_ADDRESS,
+    )
+
+    payer_transfer = factories.make_signed_transfer_for(
+        payer_channel,
+        amount,
+        UNIT_TRANSFER_SENDER,
+        target,
+        expiration,
+        UNIT_SECRET,
+    )
+
+    channel1 = factories.make_channel(
+        our_balance=amount,
+        token_address=UNIT_TOKEN_ADDRESS,
+        partner_address=target,
+    )
+    channel_map = {
+        channel1.identifier: channel1,
+        payer_channel.identifier: payer_channel,
+    }
+    possible_routes = [factories.route_from_channel(channel1)]
+
+    init_mediator = ActionInitMediator(
+        possible_routes,
+        factories.route_from_channel(payer_channel),
+        payer_transfer,
+    )
+
+    iteration = mediator.state_transition(
+        None,
+        init_mediator,
+        channel_map,
+        pseudo_random_generator,
+        block_number,
+    )
+
+    transfer = payer_transfer
+    secrethash = transfer.lock.secrethash
+
+    block_lock_expired = expiration + DEFAULT_NUMBER_OF_BLOCK_CONFIRMATIONS
+
+    assert secrethash in payer_channel.partner_state.secrethashes_to_lockedlocks
+
+    # Reveal secret just before the lock expires
+    secret_reveal = ReceiveSecretReveal(UNIT_SECRET, target)
+
+    iteration = mediator.state_transition(
+        iteration.new_state,
+        secret_reveal,
+        channel_map,
+        pseudo_random_generator,
+        block_lock_expired,
+    )
+
+    # Make sure the lock was moved
+    assert secrethash not in payer_channel.partner_state.secrethashes_to_lockedlocks
+    assert secrethash in payer_channel.partner_state.secrethashes_to_unlockedlocks
```

Comment:
please, also assert on the state of the payee channel (`channel1` in this test)

Reply:
Done [here](https://github.com/raiden-network/raiden/pull/2925/files#diff-1338dac76122f31eb8ae5894300cc752R2055)